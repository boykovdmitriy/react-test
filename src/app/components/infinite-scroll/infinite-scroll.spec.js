import React from 'react';
import {InfiniteScroll} from './index'

describe('infinite scroll component', () => {
    it('render infinite scroll component', () => {
        const handleScroll = () => {
        };
        const wrapper = shallow(
            <InfiniteScroll handleScroll={handleScroll}>
                <div className='nice-object'/>
            </InfiniteScroll>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render infinite scroll and check a child element', () => {
        const handleScroll = () => {
        };
        const wrapper = shallow(
            <InfiniteScroll handleScroll={handleScroll}>
                <div className='nice-object'/>
            </InfiniteScroll>
        );
        expect(wrapper.contains(<div className='nice-object'/>)).toEqual(true);
    });

    it('render infinite scroll without handleScroll. It is required field', () => {
        expect(() => shallow(
            <InfiniteScroll>
                <div/>
            </InfiniteScroll>
        )).toThrow();
    });

    it('render infinite scroll without children. It is required field', () => {
        const handleScroll = () => {
        };
        expect(() => shallow(
            <InfiniteScroll handleScroll={handleScroll}/>
        )).toThrow();
    });
});