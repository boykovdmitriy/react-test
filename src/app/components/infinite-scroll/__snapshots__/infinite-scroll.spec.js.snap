// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infinite scroll component render infinite scroll component 1`] = `
<section>
  <div
    className="nice-object"
  />
</section>
`;
